tests/main/gadget-update-pc/task.yaml

```diff
@@ -93,15 +101,14 @@ restore: |
     . "$TESTSLIB"/store.sh
     teardown_fake_store "$BLOB_DIR"
 
-    # restore the original gadget snap
-    snap install gadget.snap
-
-    # restoring the snapshot of / does not clean up /var/snap/pc/<rev>
-    # directories that were created during the test, given that the state is
-    # restored these directories will not be accounted for
-    for i in $(seq 0 2); do
-        rm -rf "/var/snap/pc/$((START_REVISION+i))"
+    # Restore the state of the gadget snap.
+    snap revert pc --revision="$(cat original-revision.txt)"
+    rm -f original-revision.txt
+    for rev in $(seq "$START_REVISION" "$((START_REVISION + 2))") ; do
+        snap remove --revision="$rev" pc
```

Comment:
You can pass --purge here.

Reply:
I refrained from doing so because this is not the last revision of the gadget so this makes no difference.